### ISSUE TITLE
[gfx950] fp8_mqa_logits: the >=2 GiB gl.store fallback fails to compile and aborts the process

### ISSUE THREAD

**Issue by Arist12** [ROCm/aiter#5114]
## Summary

On gfx950, `fp8_mqa_logits` picks its store instruction from the size of the `logits` tensor:

https://github.com/ROCm/aiter/blob/main/aiter/ops/triton/attention/fp8_mqa_logits.py#L225-L227

```python
BUFFER_LIMIT_BYTES = 2 * 1024 * 1024 * 1024
use_buffer_load = KV.numel() * KV.element_size() < BUFFER_LIMIT_BYTES
use_buffer_store = logits.numel() * logits.element_size() < BUFFER_LIMIT_BYTES
```

At or above 2 GiB of logits the Gluon kernel takes the `gl.store` fallback in `_store_logits_block`, and **that specialization does not compile**. Because it fails inside the Triton compiler rather than at runtime, it does not raise — it aborts the process:

```
python3: /__w/triton/triton/llvm-project/llvm/include/llvm/ADT/Sequence.h:275:
llvm::iota_range<unsigned int>::iota_range(T, T, bool) [T = unsigned int]:
Assertion `Begin <= End && "Begin must be less or equal to End."' failed.
Fatal Python error: Aborted
```

A caller cannot guard this with `try/except`. In a tensor-parallel server every rank dies at once and the whole instance has to reload its weights.

## Reproducer

No model and no framework — the shape is the only variable.

```python
import sys
import torch
from aiter.ops.triton.fp8_mqa_logits import fp8_mqa_logits
from aiter.ops.triton.utils.types import get_fp8_dtypes

NQ, NK = int(sys.argv[1]), int(sys.argv[2])
NUM_HEADS = int(sys.argv[3]) if len(sys.argv) > 3 else 32
HEAD_DIM = 128
_, e4m3 = get_fp8_dtypes()
torch.manual_seed(0)

nbytes = NQ * NK * 4
print(f"q={NQ} kv={NK} heads={NUM_HEADS}  logits={nbytes:,} B  "
      f"buffer_store={'YES' if nbytes < 2**31 else 'NO (gl.store fallback)'}", flush=True)

q = torch.randn(NQ, NUM_HEADS, HEAD_DIM, device="cuda", dtype=torch.bfloat16).to(e4m3)
kv = torch.randn(NK, HEAD_DIM, device="cuda", dtype=torch.bfloat16)
amax = kv.abs().float().amax(dim=1, keepdim=True).clamp(min=1e-4)
scale = amax / torch.finfo(e4m3).max
kv_fp8 = (kv / scale).to(e4m3)
weights = torch.randn(NQ, NUM_HEADS, device="cuda", dtype=torch.float32)
ks = torch.zeros(NQ, dtype=torch.int32, device="cuda")
ke = torch.full((NQ,), NK, dtype=torch.int32, device="cuda")

logits = fp8_mqa_logits(q, kv_fp8, scale.squeeze(-1), weights, ks, ke, clean_logits=False)
torch.cuda.synchronize()
print(f"OK: logits {tuple(logits.shape)}", flush=True)
```

```bash
python repro.py 23170 23170     # 1.9999 GiB -> buffer_store -> OK
python repro.py 23171 23171     # 2.0001 GiB -> gl.store     -> aborts
```

## Three conditions, each bracketed by one unit

Run each in its own process, since the failing one aborts.

| # | condition | measured |
|---|---|---|
| 1 | `logits.numel() * 4 >= 2**31` — selects the fallback | `23170 23170` returns; **`23171 23171`** aborts |
| 2 | `num_heads <= 32` — selects `BLOCK_M = 2` at [L206](https://github.com/ROCm/aiter/blob/main/aiter/ops/triton/attention/fp8_mqa_logits.py#L206) | `23171 23171 64` **returns**, same over-2-GiB tensor |
| 3 | `seq_len > 4096` — the other half of that same choice | `4096 200000` returns; **`4097 200000`** aborts, same 3.05 GiB tensor |

So only the multi-row store branch is affected. `_store_logits_block` is called two different ways depending on `BLOCK_M`:

https://github.com/ROCm/aiter/blob/main/aiter/ops/triton/_gluon_kernels/gfx950/attention/fp8_mqa_logits.py#L48-L67

with the `BLOCK_M == 1` path passing `logits_ptr` directly and the multi-row path passing `logits_ptr + r * stride_logits_s`. Conditions 2 and 3 both point at the second one, but I did not isolate which constexpr the compiler actually chokes on, so I would not claim more than that.

Also reproduces at exactly 2 GiB, which is worth noting because the comparison is strict `<`:

```bash
python repro.py 16384 32767     # 2,147,418,112 B -> OK
python repro.py 16384 32768     # 2,147,483,648 B, exactly 2 GiB -> aborts
```

## Why it matters in practice

These are ordinary shapes for a DSA indexer, where `logits` is `[prefill tokens x prefix length]` fp32 and grows with the product of the two. With SGLang's default `--chunked-prefill-size` of 16384 on this GPU tier, a single cold prompt of 32,768 tokens reaches 2 GiB and takes the server down — well inside the context length these models advertise. It is invisible to multi-turn traffic, because a long turn that mostly hits the prefix cache extends by few tokens and stays far under the limit; it is the first request for anyone prefilling a long document in one piece.

`index_n_heads = 32` (condition 2) and `index_head_dim = 128` are what `zai-org/GLM-5.2-FP8` and `zai-org/GLM-5.3` declare, and the indexer passes `index_n_heads` through without splitting it by TP.

SGLang is working around it by bounding the tensor so the fallback is never selected ([sgl-project/sglang#36960](https://github.com/sgl-project/sglang/pull/36960)), but the fallback stays uncompilable for any other caller.

## Environment

| | |
|---|---|
| GPU | 8x MI355X (gfx950) |
| Image | `rocm/sgl-dev:v0.5.18-rocm724-mi35x-20260829` |
| aiter | pinned at `c16d44b93a528b2a4bfd6d8d3409116d465872a9` |
| torch | 2.11.0+rocm7.2 |
| HIP | 7.2.26015 |
| Triton | 3.7.0 |

**Verified against current `main`.** The pinned checkout is behind HEAD (`4ad9983`), but `aiter/ops/triton/attention/fp8_mqa_logits.py` and `aiter/ops/triton/_gluon_kernels/gfx950/attention/fp8_mqa_logits.py` are byte-identical to `main` (md5 `ca0b822a93ccdfc25fabfa7838cccf82` and `df0484d859ded67fbdfa5bd57dea4cc4`) and no commit has touched either since 2026-08-18, so this is current-`main` behaviour.

Also reproduces identically on `rocm/sgl-dev:v0.5.18-rocm720-mi35x-20260827` (ROCm 7.2.0, torch 2.9.1, same Triton 3.7.0), so it does not look ROCm-version dependent — consistent with the assertion coming from the LLVM bundled inside Triton.

## What would help

Either making the `gl.store` fallback compile for `BLOCK_M = 2`, or — if a single store instruction genuinely cannot address past 2 GiB here — refusing the call with a clear Python-level error instead of selecting a path that aborts the process, so callers can chunk instead of dying.
